flocker/node/agents/test/blockdevicefactory.py

```diff
@@ -0,0 +1,192 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Functionality for creating ``IBlockDeviceAPI`` providers suitable for use in
+the current execution environment.
```

Comment:
It would be easier to review this, if the changes leading to this module were separated out to be reviewed separately.